### PR TITLE
Move from GCR to artifact registry

### PR DESCRIPTION
**Pull request by andrerun**
Switches the image registry used by the project output, from GCR to Artifact Registry. Also includes few minor improvements:
- Removes unnecessary version object
- Removed dead code in Makefile

### CODE REVIEW

**Review by ialidzhikov — COMMENTED** [review]
You can drop the first commit (fb3e1c372dad86977d52025b5adf379247dd02b2) from the history, it is empty one.

**Review by ialidzhikov — APPROVED** [review]
Thanks!

**Comment by ialidzhikov** [comment]
@ccwienk can you help checking this PR as you should be familiar with the required changes? Thank you in advance!
Context: This repo was private for a long time and we recently made it public. Maybe that's why you missed to adapt it when you adapted the rest of the repos.

**Comment by ialidzhikov** [comment]
@andrerun ,FYI: as part of https://github.com/gardener/gardener-custom-metrics/pull/11 I made the `concourse-ci/verify` check required. It would be best if we can get in first https://github.com/gardener/gardener-custom-metrics/pull/11, when you rebase on top of it, the corresponding check should pass.

File: .ci/pipeline_definitions [hunk 1]
```diff
@@ -16,28 +18,33 @@ gardener-custom-metrics:
...
         draft_release: ~
         options:
           public_build_logs: true
     pull-request:
       traits:
         pull-request: ~
-        component_descriptor: ~
+        component_descriptor:
+          ocm_repository_mappings:
+            - repository: europe-docker.pkg.dev/gardener-project/releases
```
Comment (ccwienk) [inline .ci/pipeline_definitions:38]: this is correct, if you always reference final versions from other components (.ci/component_descriptor callback). if you want to be able to also reference snapshot-versions, you should add `/snapshots` ocm repository as second mapping entry.

**Review by ccwienk — APPROVED** [review]

**Review by ialidzhikov — APPROVED** [review]
/lgtm

**Pull request merged**